assets/js/indexCtrl.js

```diff
@@ -47,8 +47,7 @@ module.exports = function ($scope) {
     $scope.addRefError = "";
     $scope.indexOk.addRef = false;
     $scope.indexSpin.addRef = true;
-    var url = "/reference/add",
-      user2 = $scope.addInfo.user2;
+    var url = "/reference/add";
     $scope.addRefError = $scope.validateRef($scope.addInfo);
```

Comment:
tl;dr: `$scope.addInfo.user2` gets modified by `validateRef()`, but we were still sending the old value to the server for whatever reason.